src/main/java/org/apache/maven/shared/verifier/Verifier.java

```diff
@@ -115,87 +115,195 @@
+        setMavenHome( null );
+        setDefaultCliArguments( defaultCliArguments );
     }
 
+    /**
+     *
+     * @deprecated to be removed
+     * use {@link #Verifier(String)},
+     * {@link #setSettingsFile(String)} to set settings file
+     * and {@link #setForkJvm(Boolean)} to set forkJvm status
+     *
+     * */
+    @Deprecated
     public Verifier( String basedir, String settingsFile, boolean debug, boolean forkJvm )
         throws VerificationException
     {
-        this( basedir, settingsFile, debug, forkJvm, DEFAULT_CLI_ARGUMENTS );
+        this( basedir );
+        setSettingsFile( settingsFile );
+        setDebug( debug );
+        setForkJvm( forkJvm );
     }
 
+    /**
+     *
+     * @deprecated to be removed
+     * use {@link #Verifier(String basedir)},
+     *{@link #setSettingsFile(String settingsFile)} to set settings file,
+     *{@link #setForkJvm(Boolean)} to set forkJvm status and
+     * use {@link #setDefaultCliArguments(String[] defaultCliArguments)} to set settings file
+     *
+     * */
+    @Deprecated
     public Verifier( String basedir, String settingsFile, boolean debug, boolean forkJvm, String[] defaultCliArguments )
         throws VerificationException
     {
-        this( basedir, settingsFile, debug, forkJvm, defaultCliArguments, null );
+        this( basedir );
+        setSettingsFile( basedir );
+        setDebug( debug );
+        setForkJvm( forkJvm );
+        setDefaultCliArguments( defaultCliArguments );
+        setMavenHome( null );
     }
 
+    /**
+     *
+     * @deprecated to be removed
+     * use {@link #Verifier(String basedir)},
+     * {@link #setSettingsFile(String settingsFile)} to set settings file
+     * and {@link #setMavenHome(String mavenHome)}  to set maven home
+     *
+     * */
+    @Deprecated
     public Verifier( String basedir, String settingsFile, boolean debug, String mavenHome )
             throws VerificationException
     {
-        this( basedir, settingsFile, debug, null, DEFAULT_CLI_ARGUMENTS, mavenHome );
+        this( basedir );
+        setForkJvm( null );
+        setDebug( debug );
+        setSettingsFile( settingsFile );
+        setMavenHome( mavenHome );
     }
 
+    /**
+     *
+     * @deprecated to be removed
+     * use {@link #Verifier(String basedir)}
+     * use {@link #setSettingsFile(String settingsFile)} to set settings file
+     * use {@link #setMavenHome(String mavenHome)}  to set maven home
+     * use {@link #setDefaultCliArguments(String[] defaultCliArguments)} to set settings file
+     *
+     * */
+    @Deprecated
     public Verifier( String basedir, String settingsFile, boolean debug, String mavenHome,
                      String[] defaultCliArguments )
         throws VerificationException
     {
-        this( basedir, settingsFile, debug, null, defaultCliArguments, mavenHome );
+        this( basedir );
+        setSettingsFile( settingsFile );
+        setDebug( debug );
+        setForkJvm( null );
```

Comment:
?

Reply:
setting null for ForkJvm is also unnecessary, I will remove this as well.